### ISSUE TITLE
lto1: internal compiler error: Segmentation fault

### ISSUE THREAD

**Issue by omg-itsme** [vtnerd/monero-lws#28]
[93%] Linking CXX executable monero-lws-daemon
during IPA pass: icf
lto1: internal compiler error: Segmentation fault
Please submit a full bug report,
with preprocessed source if appropriate.
See <file:///usr/share/doc/gcc-9/README.Bugs> for instructions.
lto-wrapper: fatal error: /usr/bin/c++ returned 1 exit status
compilation terminated.
/usr/bin/ld: error: lto-wrapper failed
collect2: error: ld returned 1 exit status
make[2]: *** [src/CMakeFiles/monero-lws-daemon.dir/build.make:153: src/monero-lws-daemon] Error 1
make[1]: *** [CMakeFiles/Makefile2:204: src/CMakeFiles/monero-lws-daemon.dir/all] Error 2
make: *** [Makefile:130: all] Error 2

Need a fix

**Comment by selsta** [comment]
Try to update your compiler

**Comment by vtnerd** [comment]
I agree with @selsta , an ICEd compiler could be the result of incorrect code, but the compiler should generate a proper error instead of crashing.
